packages/core/src/shared/vscode/env.ts

```diff
@@ -68,13 +68,19 @@ export { extensionVersion }
  * @param throwWhen Throw if minimum vscode is equal or later than this version.
  */
 export function isMinVscode(throwWhen?: string): boolean {
-    const minVscode = getMinVscodeVersion()
-    if (throwWhen && semver.gte(minVscode, throwWhen)) {
-        throw Error(`Min vscode ${minVscode} >= ${throwWhen}. Delete or update the code that called this.`)
+    if (throwWhen) {
+        throwIfMinVscodeGte(throwWhen)
     }
     return vscode.version.startsWith(getMinVscodeVersion())
 }
 
+export function throwIfMinVscodeGte(targetVersion: string): void | never {
```

Comment:
`isMinVscode` is doing multiple things, but that can be fixed by making its arg a dict , so that calls look like:

    isMinVscode({ throwWhen: ... })